src/llama-vocab.cpp

```diff
@@ -520,7 +520,7 @@ struct llm_tokenizer_bpe {
 
             // build token(s)
             while (!work_queue.empty()) {
-                auto bigram = work_queue.top();
+                auto bigram = std::move(const_cast<llm_bigram_bpe&>(work_queue.top()));
```

Comment:
Prefer to avoid `const_cast`. Better to switch to a heap container that supports move semantic

Reply:
I've added a commit that tries to address this. Was this similar to what you had in mind?